packages/api/src/Base.ts

```diff
@@ -374,12 +374,27 @@ export default abstract class ApiBase<CodecResult, SubscriptionResult> implement
   }
 
   private async loadMeta (): Promise<boolean> {
+    const { prebundles = {} } = this._options;
     // only load from on-chain if we are not a clone (default path), alternatively
     // just use the values from the source instance provided
     if (!this._options.source || !this._options.source._isReady) {
-      this._runtimeMetadata = await this._rpcBase.state.getMetadata();
-      this._runtimeVersion = await this._rpcBase.chain.getRuntimeVersion();
-      this._genesisHash = await this._rpcBase.chain.getBlockHash(0);
+      [this._genesisHash, this._runtimeVersion] = await Promise.all([
+        this._rpcBase.chain.getBlockHash(0),
+        this._rpcBase.chain.getRuntimeVersion()
+      ]);
+      const prebundlesKey = `${this._genesisHash}-${(this._runtimeVersion as RuntimeVersion).specVersion}`;
+      if (prebundlesKey in prebundles) {
+        try {
+          const rpcData = prebundles[prebundlesKey];
+          const metadata = new Metadata(rpcData);
+          this._runtimeMetadata = metadata;
+          l.warn('Experimental feature. Developers to be carefully while adding prebundled metadata');
```

Comment:
The issue with the warning here as it stands (since it is not tied to anything) is that it will pop up for all users - which is going to create more issues for mom-and-pop asking what is wrong ;) Not sure how to get around it at all, so I'd rather err on the side of not having it.

Reply:
done